### PR TITLE
Fixes to address dropped connectivity

### PR DESCRIPTION
**Pull request by iicaisin1**
- added TcpHelper to detect better if the tcpclient is connected
- fixed the issue with unreleased lock in the confirmationsendservice
- added the default styling config (included spaces as default indentation style). Project owner can enhance it further

### CODE REVIEW

File: src/TelegramClient.Core/Helpers/TcpClientHelper.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using System.Net.Sockets;
+
+namespace TelegramClient.Core.Helpers
+{
+    public static class TcpClientHelper
+    {
+        public static bool IsConnected(this TcpClient client)
```
Comment (vik-borisov) [inline src/TelegramClient.Core/Helpers/TcpClientHelper.cs]: move method to TcpService class.
In this case, the extension methods are not needed

**Comment by iicaisin1** [comment]
Apparently, this does not address all the issues with losing connection. Still reviewing other possible issues...

**Comment by vik-borisov** [comment]
@iicaisin1, please fix codacy's and my comments

File: src/TelegramClient.Core/Network/Tcp/TcpTransport.cs [hunk 2]
```diff
@@ -33,67 +34,85 @@ public void Dispose()
...
+                    throw new InvalidOperationException("invalid checksum! skip");
+                }
+
+                return body;
+            }
+            catch
             {
-                throw new InvalidOperationException("invalid checksum! skip");
+                var buffer = new byte[4];
+                while (stream.CanRead && stream.DataAvailable)
```
Comment (iicaisin1) [inline src/TelegramClient.Core/Network/Tcp/TcpTransport.cs:109]: @vik-borisov,
I wrapped it into a try catch and drained the tcp stream to get back to a good state. This is an ugly implementation and skips the malformed message without a retry. Maybe you have a better idea what can be done here

**Comment by iicaisin1** [comment]
Although the commits in this PR address a bunch of issues, I just noticed another issue that is reproducible. If the response for a request comes in a malformed state (e.g. i have seen on sparse occasions when the response could not be decoded), the request will be in the await state forever.

**Pull request merged**